solidity/contracts/Coordinator.sol

```diff
@@ -90,11 +96,12 @@ contract Coordinator is ChainlinkRequestInterface, CoordinatorInterface {
   {
     bytes32 requestId = keccak256(abi.encodePacked(_sender, _nonce));
     require(callbacks[requestId].cancelExpiration == 0, "Must use a unique ID");
-    callbacks[requestId] = Callback(
-      _amount,
-      _callbackAddress,
-      _callbackFunctionId,
-      uint64(now.add(5 minutes)));
+
+    callbacks[requestId].sAId = _sAId;
+    callbacks[requestId].amount = _amount;
+    callbacks[requestId].addr = _callbackAddress;
+    callbacks[requestId].functionId = _callbackFunctionId;
+    callbacks[requestId].cancelExpiration = uint64(now.add(EXPIRY_TIME));
```

Comment:
I had to leave `EXPIRY_TIME` here for the moment instead of the SA's actual expiration. We've hit the stack limit for this method. Will be fixed when callbacks are migrated to commitments in [#163682750](https://www.pivotaltracker.com/story/show/163682750).